src/librustc_mir/interpret/place.rs

```diff
@@ -386,8 +386,9 @@ where
         let dl = &self.tcx.data_layout;
         Ok((0..len).map(move |i| {
             let ptr = base.ptr.ptr_offset(i * stride, dl)?;
+            let align = base.align.restrict_for_offset(i * stride);
```

Comment:
Can this ever make a difference?

Reply:
I agree it seems fundamentally more correct though.

We now have three methods (or more?) that do "add offset to an `MemPlace`, adjust alignment". Can you turn that into a method on `MemPlace`?